File: 0019. 删除链表的倒数第N个节点/链表.swift
// 8ms
class Solution {
    func removeNthFromEnd(_ head: ListNode?, _ n: Int) -> ListNode? {
        var len = 0, front = ListNode(-1, head), current = front
        while let node = current.next {
            len += 1
            current = node
        }
        var index = len - n - 1
        current = front
        (-1..<index).forEach { _ in current = current.next! }
        var next = current.next!
        current.next = next.next
        next.next = nil
        return front.next
    }
}

File: 0203. 移除链表元素/链表.js
// 136ms
var removeElements = function(head, val) {
    head = new ListNode(0, head)
    let current = head
    while (current.next !== null) {
        if (current.next.val === val) {
            let del = current.next
            current.next = del.next
            del.next = null
        } else {
            current = current.next
        }
    }
    return head.next
};